cloud/blockstore/vhost-server/critical_event.cpp

```diff
@@ -0,0 +1,48 @@
+#include "critical_event.h"
+
+#include <cloud/storage/core/libs/diagnostics/logging.h>
+
+#include <util/string/builder.h>
+#include <util/system/spinlock.h>
+
+namespace NCloud::NBlockStore::NVHostServer {
+
+////////////////////////////////////////////////////////////////////////////////
+
+TLog Log;
+TCriticalEvents CriticalEventsAccumulator;
+TAdaptiveLock Guard;
+
+void SetCriticalEventsLog(TLog log)
+{
+    Log = std::move(log);
+}
+
+void ReportCriticalEvent(TString sensorName, TString message)
+{
+    if (Log.IsOpen()) {
+        TStringBuilder fullMessage;
+        fullMessage << "CRITICAL_EVENT:" << sensorName;
+        if (message) {
+            fullMessage << ":" << message;
+        }
+        STORAGE_ERROR(fullMessage);
+    }
+
+    with_lock (Guard) {
+        CriticalEventsAccumulator.emplace_back(
```

Comment:
maybe put some limit here on the size of vector ... otherwise in some weird situation we restart nbs and vhost server keeps living and consumes too much memory before it dies. Or for some weird reason we don't report statistics and have flood of ciritical events that will consume all memory